src/store/Actions.ts

```diff
@@ -31,44 +30,59 @@ export enum AsyncType {
  * InnterAction are internal sync actions.
  */
 export type InnerAction =
-  | { type: InnerType.ErrorFetch; message: string }
   | { type: InnerType.StartSignUp }
   | { type: InnerType.FinishSignUp; sessionInfo: UserSessionInfo };
 
 /**
  * OuterAction are external sync actions.
  */
-export type OuterAction = { type: OuterType.Increment };
+export type OuterAction =
+  | { type: OuterType.SetError; message: string }
+  | { type: OuterType.ClearError };
 
 /**
  * Action represents all possible sync actions.
  */
 export type Action = InnerAction | OuterAction;
 
+type Callback<T = any> = (arg: T) => void;
+
 /**
  * AsyncAction represents all possible async actions.
  */
 export type AsyncAction = {
   type: AsyncType.SignUp;
   username: string;
   email: string;
+  callback?: Callback<UserSessionInfo>;
```

Comment:
i'm thinking it'll be useful to let all async actions take a callback as an optional last arg